### PR TITLE
doc: add roadmap doc for better community building

### PR DESCRIPTION
**Pull request by Starnop**
Signed-off-by: Starnop <starnop@163.com>

### Ⅰ. Describe what this PR did

`ROADMAP` doc is common in open source projects , and it is also a friendly delivery that enables contributors to better participate in the community. For Dragonfly, We hope that contributors are willing to participate with it better. 

### Ⅱ. Does this pull request fix one issue?

None.

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)
None.

### Ⅴ. Special notes for reviews
Looking forward to your further Suggestions.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -2,34 +2,102 @@
 title: "Roadmap"
 ---
 
-## CNCF Ecosystem
+# Dragonfly Roadmap
 
-- Deploy SuperNode using Helm in Kubernetes
-- Deploy dfget & dfdaemon using DaemonSet in Kubernetes
-- Integration with Harbor: preheat image feature
```
Comment (ZYecho) [inline docs/roadmap.md]: Maybe add a link for this feature? since people may don't know it, such as me.

File: docs/roadmap.md [hunk 2]
```diff
@@ -2,34 +2,102 @@
...
+* Different encryption algorithm in data transmission.
+
+### Function Enhancement
+
+Add more advanced functionality on the basis of previous versions.
+
+* Support multi-registrys.
+* Support supernode cluster. Cluster the SuperNode to decrease possibility of failure.
+* Supernode support storage driver.
+* Use IPFS to share block datas between SuperNodes.
```
Comment (ZYecho) [inline docs/roadmap.md]: for the ipfs, seems as above.

**Comment by fengzixu** [comment]
In my opinion, I think that there are some functions needed to be considered:

1. We should add the basic mechanism of monitoring of Dragonfly before it was published, so that the users can know situations about the processing of Dragongly. The Grafana and Prometheus are valuable to be considered. And https://github.com/grpc-ecosystem/go-grpc-prometheus is also an important repo for us.

2. And I found that the community has a plan: make the Dragonfly as a cloud-native application. That is a very awesome choice. Using the `helm` to install the dfdaemon and using `daemonset` to deploy the dfget is the first step. If we want to create a high-available Dragonfly, we perhaps use "CRD" and "Operator" to create and maintain it in Kubernetes cluster. Fortunately, I am very good at this area. I have developed the Redis-Operator for my company. If the community want to do this, let me lead it.

3. Now that we have made a plan and created some checkpoint for the GA of Dragonfly. I considered that we should open the weekly meeting after core works are assigned to contributors. In the meeting, we can confirm what tasks were delayed, what the schedule of Dragonfly at this week. It can help us to guarantee the tasks of the specific version of Dragonfly process successfully。

WDYT @Starnop @allencloud

**Comment by ZYecho** [comment]
> 2\. If we want to create a high-available Dragonfly

Interested in part 2, I can help to finish work related kubernetes. @fengzixu if you have any plan, please cc me, thanks. ^-^

File: docs/roadmap.md (hunk 1 above)
Comment (Starnop) [reply docs/roadmap.md]: Sounds great.

**Comment by Starnop** [comment]
> Now that we have made a plan and created some checkpoint for the GA of Dragonfly. I considered that we should open the weekly meeting after core works are assigned to contributors. In the meeting, we can confirm what tasks were delayed, what the schedule of Dragonfly at this week. It can help us to guarantee the tasks of the specific version of Dragonfly process successfully。

As for weekly meeting, we can't wait to do it to make our community better. At present, this matter has been in the plan,  we will keep you abreast of the progress in the [Dingtalk](https://github.com/dragonflyoss/Dragonfly#community) Group and the github community.

**Comment by allencloud** [comment]
Could you add a table at the front of the file to clearly identify the relationship among release, main part and deadline? @Starnop

**Comment by allencloud** [comment]
LGTM

**Pull request merged**